runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/MetricsToPerStepNamespaceMetricsConverter.java

```diff
@@ -52,16 +52,16 @@ private static Optional<MetricValue> convertCounterToMetricValue(
       return Optional.empty();
     }
 
-    BigQuerySinkMetrics.ParsedMetricName labeledName =
+    Optional<BigQuerySinkMetrics.ParsedMetricName> labeledName =
```

Comment:
nit: you can do 

```
return BigQuerySinkMetrics.parseMetricName(metricName.getName())
// filter for basenames !empty 
.filter(labeledName -> !labeledName.isEmpty())
// Only called if above predicate is true
.map(labeledName -> new MetricValue()
            .setMetric(labeledName.getBaseName())
            .setMetricLabels(labeledName.getMetricLabels())
            .setValueInt64(value)));
```

if `BigQuerySinkMetrics.parseMetricName(metricName.getName())` returns null or the filter predicate fails will default to Optional.empty().

this is more of a style thing

Reply:
Done.